scanpipe/models.py

```diff
@@ -1600,6 +1601,18 @@ def path_pattern(self, pattern):
         """Resources with a path that match the provided ``pattern``."""
         return self.filter(path__regex=posix_regex_to_django_regex_lookup(pattern))
 
+    def has_directory_content_fingerprint(self):
+        """
+        Resources that have the key `directory_content` set in the `extra_data`
+        field.
+        """
+        return self.filter(~Q(extra_data__directory_content=""))
+
+    def paginated(self, per_page=5000):
+        """Iterate over a (large) QuerySet by chunks of ``per_page`` items."""
+        for page in Paginator(self, per_page=per_page):
+            yield page.object_list
```

Comment:
Could you add some content in the docstring about why this is different from using the `queryset.iterator` method and about the benefit/context for using this?